### DISCUSSION TITLE
How to divide large amount of entities in a collection?

### DISCUSSION THREAD

**Discussion by daidoli** [milvus-io/milvus#29129] [Q&A and General discussion] [answered]
I created 11M entities in a collection. It takes 40GB memory but only loads 50% entities. There is no enough memory to load entire collection and can't perform query or search. Is there any way to divide an existing collection? I'm using Milvus v2.3.2 standalone.

**Comment by yhmo** [comment, marked as answer]
If your vector dimension is 1536, for 11M entities, the original data size is `11M*1536*4bytes = 68GB`.
If you choose IVF_FLAT/HNSW index, the index size is a bit larger than the original data size, query nodes will require 70GB+ to load the collection.

If you want to search in the collection, milvus requires the collection in memory even you divide the data to partitions.
To reduce memory usage, you can use IVF_SQ8/IVF_PQ or DISKANN index, these indexes only require 20%~30% memory space. For example, 1536dim, 11M entities, IVF_SQ8 requires `68GB*25%=17GB` memory space.

**Reply by daidoli** [reply]
I changed index from IVF_FLAT to IVF_PQ. It works. Now the required memory size are 40GB (dim is 1024). Thanks.
